skore/src/skore/sklearn/_comparison/metrics_accessor.py

```diff
@@ -221,11 +221,11 @@ def _compute_metric_scores(
         total_estimators = len(self._parent.reports_)
         progress.update(main_task, total=total_estimators)
 
-        if self._parent._reports_type == "CrossValidationReport":
-            # Ensure that there is only one Progress instance.
-            # EstimatorReports don't have `_parent_progress`
-            for report in self._parent.reports_:
+        reports_to_cleanup = []
+        for report in self._parent.reports_:
+            if hasattr(report, "_parent_progress"):
                 report._parent_progress = progress
+                reports_to_cleanup.append(report)
```

Comment:
```suggestion
        for report in self._parent.reports_:
            if hasattr(report, "_parent_progress"):
                report._parent_progress = progress
```
`reports_to_cleanup` is never used.